src/lib.rs

```diff
@@ -11,6 +11,898 @@
  * SPDX-License-Identifier: Apache-2.0
  ********************************************************************************/
 
+use cxx::{let_cxx_string, UniquePtr};
+use lazy_static::lazy_static;
+use log::{error, info, trace};
+use std::collections::HashSet;
+use std::path::{Path, PathBuf};
+use std::sync::Arc;
+use std::thread;
+use std::time::Duration;
+use tokio::runtime::Builder;
+use tokio::sync::mpsc::{channel, Receiver, Sender};
+use tokio::sync::{oneshot, RwLock};
+use up_rust::{UCode, UListener, UMessage, UMessageType, UStatus, UUri, UUID};
+use vsomeip_sys::extern_callback_wrappers::MessageHandlerFnPtr;
+use vsomeip_sys::glue::{
+    make_application_wrapper, make_runtime_wrapper, ApplicationWrapper, RuntimeWrapper,
+};
+use vsomeip_sys::safe_glue::{
+    get_message_payload, get_pinned_application, get_pinned_message_base, get_pinned_runtime,
+    register_message_handler_fn_ptr_safe, request_single_event_safe,
+};
+use vsomeip_sys::vsomeip;
+use vsomeip_sys::vsomeip::{ANY_MAJOR, ANY_MINOR};
+
 pub mod transport;
+use transport::CLIENT_ID_APP_MAPPING;
+
+mod message_conversions;
+use message_conversions::convert_umsg_to_vsomeip_msg;
+
+mod determinations;
+use determinations::{
+    create_request_id, find_app_name, retrieve_session_id, split_u32_to_u16, split_u32_to_u8,
+};
+
+mod vsomeip_config;
+
+const UP_CLIENT_VSOMEIP_TAG: &str = "UPClientVsomeip";
+const UP_CLIENT_VSOMEIP_FN_TAG_NEW_INTERNAL: &str = "new_internal";
+const UP_CLIENT_VSOMEIP_FN_TAG_APP_EVENT_LOOP: &str = "app_event_loop";
+const UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL: &str = "register_listener_internal";
+const UP_CLIENT_VSOMEIP_FN_TAG_UNREGISTER_LISTENER_INTERNAL: &str = "unregister_listener_internal";
+const UP_CLIENT_VSOMEIP_FN_TAG_SEND_INTERNAL: &str = "send_internal";
+const UP_CLIENT_VSOMEIP_FN_TAG_INITIALIZE_NEW_APP_INTERNAL: &str = "initialize_new_app_internal";
+const UP_CLIENT_VSOMEIP_FN_TAG_START_APP: &str = "start_app";
+
+// TODO: Revisit what authority to attach, if any, to the remote mE. For now, just assign "me_authority"
```

Comment:
@stevenhartley -- SOME/IP doesn't have a concept of an authority, so here I've chosen to attach this.

Any thoughts?

Reply:
Authority in the case of SOME/IP devices will refer the IP addresses of the devices that the mechatronics services run on. We are either going to use the static IP address or the hostnames (TBD) but we need to fetch that information (it is sen din the offer_service() AFAIK).